### PR TITLE
Implemented ContentCreate/UserCreate gateway

### PR DESCRIPTION
**Pull request by bdunogier**
When a ContentCreate for a User is POSTed, it is handled by the ContentUserGateway.

It creates the user, using the User API, from the request, and returns the created user.
No controller is used in the process, it all happens in the event.

### CODE REVIEW

**Comment by bdunogier** [comment]
Ping @dpobel @andrerom.

File: Rest/ContentUserGateway.php [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        if (!$request->attributes->get('is_rest_request')) {
+            return;
+        }
+
+        if (!$contentTypeHeaderValue = $request->headers->get('content-type')) {
```
Comment (andrerom) [inline Rest/ContentUserGateway.php]: we tend to prefer parentheses for these right?
`if (!($contentTypeHeaderValue = $request->headers->get('content-type'))) {`

File: Rest/ContentUserGateway.php [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        if (!$message->body) {
+            return;
+        }
+        $result = $this->restInputDispatcher->parse($message);
+        if (!$result instanceof RestContentCreateStruct) {
+            return;
+        }
+
+        // Not a user
+        if (($createData = $this->mapContentCreateToUserCreate($result)) === false) {
```
Comment (andrerom) [inline Rest/ContentUserGateway.php]: nitpick, but  `$userCreateData`

**Comment by andrerom** [comment]
+1, seems like an ok approach

File: Rest/ContentUserGateway.php (hunk 1 above)
Comment (bdunogier) [inline Rest/ContentUserGateway.php]: You're right.

**Comment by bdunogier** [comment]
I have been able to test it, and it "works", to some extent.

Problem A: publish will generate an error: creating will work, but will also publish directly (that's what `UserService::createUser()` does. We could remove the "Publish" button, change the label of "Save" to "Publish", and make it exit... @dpobel ?

Problem B: save first, then publish won't work, as it will try to _update_ the user. I have only implemented support for _create_.

**Comment by bdunogier** [comment]
Agreed with @dpobel that it was "better", and could go in as is. Will be improved when needed.

**Comment by bdunogier** [comment]
ezrobot is drunk, I've updated this hours ago. Merging :-)

**Pull request merged**